domain/modeldefaults/state/state.go

```diff
@@ -371,51 +520,55 @@ ON CONFLICT(cloud_uuid, key) DO UPDATE
     AND key = excluded.key;
 `, cloudDefaultValue{})
 	if err != nil {
-		return errors.Trace(err)
+		return errors.Capture(err)
 	}
 
 	return db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
 		for k, v := range updateAttrs {
-			err := tx.Query(ctx, upsertStmt, cloudDefaultValue{UUID: cld.UUID, Key: k, Value: v}).Run()
+			err := tx.Query(ctx, upsertStmt, cloudDefaultValue{UUID: cloudUUID.String(), Key: k, Value: v}).Run()
 			// The cloud UUID has previously been checked. This allows us to avoid having to use RunAtomic.
 			if database.IsErrConstraintForeignKey(err) {
-				return interrors.Errorf("cloud %q not found", cloudUUID).Add(clouderrors.NotFound)
+				return errors.Errorf("cloud %q not found", cloudUUID).Add(clouderrors.NotFound)
 			} else if err != nil {
-				return interrors.Errorf("updating %q cloud default keys: %w", cloudUUID, err)
+				return errors.Errorf("updating cloud %q default keys: %w", cloudUUID, err)
 			}
 		}
 		return nil
 	})
 }
 
-// DeleteCloudDefaults deletes the specified cloud default
-// config values for the provided keys if they exist.
+// DeleteCloudDefaults will delete the specified default keys from the cloud if
+// they exist. If the cloud does not exist an error satisfying
+// [clouderrors.NotFound] will be returned.
 func (s *State) DeleteCloudDefaults(
 	ctx context.Context,
 	cloudUUID cloud.UUID,
 	removeAttrs []string,
 ) error {
 	db, err := s.DB()
 	if err != nil {
-		return errors.Trace(err)
+		return errors.Capture(err)
 	}
 
-	cld := dbCloud{UUID: cloudUUID.String()}
+	cld := cloudUUIDValue{UUID: cloudUUID.String()}
 	toRemove := attrs(removeAttrs)
 
 	deleteStmt, err := s.Prepare(`
 DELETE FROM cloud_defaults
 WHERE key IN ($attrs[:])
-AND cloud_uuid = $dbCloud.uuid;
+AND cloud_uuid = $cloudUUIDValue.uuid;
 `, toRemove, cld)
 	if err != nil {
-		return errors.Trace(err)
+		return errors.Capture(err)
 	}
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		if err := s.checkCloudExists(ctx, tx, cloudUUID); err != nil {
```

Comment:
Do we need this check for DeleteCloudDefaults?

Reply:
My personal belief is we should. We should be strict and correct at the lowest levels of Juju. If should be the decision of a higher level user facing component as to if this should be considered something the end caller cares about.

We have many client to Juju now with different concerns about correctness.